src/fake_useragent/utils.py

```diff
@@ -1,21 +1,64 @@
 import json
 import sys
+from enum import Enum
+from typing import TypedDict, Union
 
 # We need files() from Python 3.10 or higher
 if sys.version_info >= (3, 10):
     import importlib.resources as ilr
 else:
     import importlib_resources as ilr
 
+
 from fake_useragent.log import logger
 
 str_types = (str,)
 
 
+class DeviceType(str, Enum):
+    PC = "pc"
+    MOBILE = "mobile"
+    TABLET = "tablet"
+
+
+class Browser(str, Enum):
+    EDGE = "edge"
+    CHROME = "chrome"
+    DUCKDUCKGO = "DuckDuckGo Mobile"
+    SAFARI = "safari"
+    FIREFOX = "firefox"
+
+
+class OS(str, Enum):
+    ANDROID = "android"
+    WIN10 = "win10"
+    IOS = "ios"
+    MACOS = "macos"
+    LINUX = "linux"
+
+
+class BrowserUserAgentData(TypedDict):
+    useragent: str
+    """The useragent string."""
+    percent: float
+    """Sampling probability for this useragent when random sampling. Currently has no effect."""
+    type: DeviceType
+    """The device type for this useragent."""
+    system: str
+    """System name for the useragent."""
+    browser: Browser
+    """Browser name for the useragent."""
+    version: float
+    """Version of the browser."""
+    os: str
```

Comment:
```suggestion
    os: OS
```

I think you then also want to use the class?